### PR TITLE
Add release notes for 10.0.5 #975

### PR DESCRIPTION
**Pull request by jlprat**
Issue: #975
Provide release notes for release 10.0.5

### CODE REVIEW

File: docs/src/main/paradox/release-notes.md [hunk 1]
```diff
@@ -1,5 +1,38 @@
...
+
+##### akka-http-core
+ * Prevent longer-than-needed lingering streams by fixing DelayCancellationStage ([#945](https://github.com/akka/akka-http/issues/945))
+
+##### akka-http
+ * Avoid redirect-loop when redirectToNoTrailingSlashIfPresent was used for root path ([#878](https://github.com/akka/akka-http/issues/878))
+
+### Compatibility notes
+
+This version of Akka HTTP must be used with Akka in version at-least 2.4.17, however it is also compatible with Akka 2.5, which has just released its Release Candidate 1.
```
Comment (jlprat) [inline docs/src/main/paradox/release-notes.md]: Added this note, as it's important to note that this version must be used with Aka 2.4.17 or higher

File: docs/src/main/paradox/release-notes.md [hunk 2]
```diff
@@ -1,5 +1,38 @@
 # Release Notes
 
+## 10.0.5
+
+See the [announcement](http://akka.io/news/2017/03/17/akka-http-10.0.5-released.html) and
+closed tickets on the [10.0.5 milestone](https://github.com/akka/akka-http/milestone/22?closed=1).
+
+This is the fifth maintenance release of the Akka Http 10.0 series. It is primarily aimed at stability aligning the internals with the upcoming Akka 2.5 release. These steps are also the groundwork to enable Play to make use of Akka HTTP and the new Akka Streams materializer in the upcoming Play 2.6.
```
Comment (jonas) [inline docs/src/main/paradox/release-notes.md]: Similar to the announcement there's too much inconsistency for me: `Akka Http` or `Akka HTTP`, pick one, preferably the latter. ;-)

Comment (jlprat) [inline docs/src/main/paradox/release-notes.md]: Yes, this is a direct copy of the announcement. I can fix it in here ;)

File: docs/src/main/paradox/release-notes.md (hunk 1 above)
Comment (ktoso) [inline docs/src/main/paradox/release-notes.md]: Yeap, thanks

**Comment by jrudolph** [comment]
Late thanks for doing the release, @jlprat and @ktoso! :fireworks:

**Comment by jlprat** [comment]
Thanks, but I did only some doc copy-pasting, @ktoso did the whole thing!

**Pull request merged**